homeassistant/helpers/config_validation.py

```diff
@@ -499,11 +499,22 @@ def validator(value):
 
 # Schemas
 
+COMPONENT_SCHEMA = vol.Schema({
```

Comment:
I would expect it to be structured like this:

```python
PLATFORM_SCHEMA_BASE = vol.Schema({
    vol.Required(CONF_PLATFORM): string,
    vol.Optional(CONF_SCAN_INTERVAL): time_period
})

PLATFORM_SCHEMA = PLATFORM_SCHEMA_BASE.extend({
}, extra=vol.ALLOW_EXTRA)
```

And components will import `PLATFORM_SCHEMA`, platforms will import `PLATFORM_SCHEMA_BASE`

Reply:
If so, every single platform module will have to be updated though.
Also, platforms are supposed to extend the component schema if I understood @MartinHjelmare correctly.